cpp/dolfinx/fem/assemble_scalar_impl.h

```diff
@@ -86,10 +85,11 @@ T assemble_exterior_facets(
   std::vector<double> coordinate_dofs(3 * num_dofs_g);
 
   // Iterate over all facets
-  for (std::size_t index = 0; index < facets.size(); ++index)
+  assert(facets.size() % 2 == 0);
+  for (std::size_t index = 0; index < facets.size() / 2; ++index)
```

Comment:
`for (std::size_t index = 0; index < facets.size(); index += 2)` and simplifications below.